src/directives/if-defined.ts

```diff
@@ -25,7 +25,8 @@ const previousValues = new WeakMap<Part, unknown>();
 export const ifDefined = directive((value: unknown) => (part: Part) => {
   const previousValue = previousValues.get(part);
 
-  if (value === undefined && part instanceof AttributePart) {
+  if (value === undefined && part instanceof AttributePart &&
```

Comment:
Not sure what you're asking.  The whole directive can't be skipped if `previousValue === undefined`, because we might need to call `setValue` if `value` is defined. And we can't just call `setValue` because `previousValue === undefined`, since we don't ever want to set `undefined`.  So there are 3 cases.